packages/valory/skills/check_stop_trading_abci/rounds.py

```diff
@@ -0,0 +1,154 @@
+
+"""This module contains the rounds for the check stop trading ABCI application."""
+
+from abc import ABC
+from enum import Enum
+from typing import Dict, Optional, Set, Tuple, Type, cast
+
+from packages.valory.contracts.service_staking_token.contract import StakingState
+from packages.valory.skills.abstract_round_abci.base import (
+    AbciApp,
+    AbciAppTransitionFunction,
+    AbstractRound,
+    AppState,
+    BaseSynchronizedData,
+    CollectSameUntilThresholdRound,
+    CollectionRound,
+    DegenerateRound,
+    DeserializedCollection,
+    get_name,
+)
+from packages.valory.skills.check_stop_trading_abci.payloads import CheckStopTradingPayload
+
+
+class Event(Enum):
+    """Event enumeration for the check stop trading skill."""
+
+    DONE = "done"
+    NONE = "none"
+    ROUND_TIMEOUT = "round_timeout"
+    NO_MAJORITY = "no_majority"
+    SKIP_TRADING = "skip_trading"
+
+
+class SynchronizedData(BaseSynchronizedData):
+    """Class to represent the synchronized data.
+
+    This data is replicated by the tendermint application.
+    """
+
+    def _get_deserialized(self, key: str) -> DeserializedCollection:
+        """Strictly get a collection and return it deserialized."""
+        serialized = self.db.get_strict(key)
+        return CollectionRound.deserialize_collection(serialized)
+
+    @property
+    def stop_trading(self) -> bool:
+        """Get if the service must stop trading."""
+        return bool(self.db.get("stop_trading", False))
+
+    @property
+    def participant_to_selection(self) -> DeserializedCollection:
+        """Get the participants to selection round."""
+        return self._get_deserialized("participant_to_selection")
+
+
+class CheckStopTradingRound(CollectSameUntilThresholdRound):
+    """A round for checking stop trading conditions."""
+
+    payload_class = CheckStopTradingPayload
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    none_event = Event.NONE
+    no_majority_event = Event.NO_MAJORITY
+    selection_key = (get_name(SynchronizedData.stop_trading),)
+    collection_key = get_name(SynchronizedData.participant_to_selection)
+
+    def end_block(self) -> Optional[Tuple[SynchronizedData, Enum]]:
+        """Process the end of the block."""
+        res = super().end_block()
+        if res is None:
+            return None
+
+        synced_data, event = cast(Tuple[SynchronizedData, Enum], res)
+        stop_trading_payload = self.most_voted_payload
+
+        if event == Event.DONE and stop_trading_payload == True:
+            return synced_data, Event.SKIP_TRADING
+
+        return synced_data, event
+
+
+class FinishedCheckStopTradingRound(DegenerateRound, ABC):
+    """A round that represents check stop trading has finished."""
+
+
+class FinishedCheckStopTradingWithSkipTradingRound(DegenerateRound, ABC):
+    """A round that represents check stop trading has finished with skip trading."""
```

Comment:
It is not necessary to prefix with `FinishedCheckStopTrading`. Could you please change the names to something more intuitive?

Reply:
Fixed.